python/pyarrow/tests/test_fs.py

```diff
@@ -1052,6 +1052,50 @@ def test_s3_options(monkeypatch):
         )
 
 
+@pytest.mark.s3
+def test_s3_proxy_options(monkeypatch):
+    from pyarrow.fs import S3FileSystem
+
+    # Avoid wait for unavailable metadata server in ARN role example below
+    monkeypatch.setenv("AWS_EC2_METADATA_DISABLED", "true")
+
+    # Check dict case for 'proxy_options'
+    fs1 = S3FileSystem(proxy_options={'scheme': 'http', 'host': 'localhost',
+                                      'port': 8999})
+    assert isinstance(fs1, S3FileSystem)
+    assert pickle.loads(pickle.dumps(fs1)) == fs1
+
+    # Check str case for 'proxy_options'
+    fs2 = S3FileSystem(proxy_options='http://localhost:9000')
+    assert isinstance(fs2, S3FileSystem)
+    assert pickle.loads(pickle.dumps(fs2)) == fs2
+
```

Comment:
Can you check inequality between two different set of proxy options?
Can you also check equality of equivalent proxy options create from a dict or a URI?

Reply:
Sure, adding tests for all combinations to be comprehensive.

Reply:
@pitrou Can you check that all the tests are there now?

Reply:
Or did you mean add tests for the pickle-unpickle case?

Reply:
Unless I'm missing something, you may add a test with the 'https' scheme as well? :-)

Reply:
Sure, will just convert `proxy_opts_2` to use `https`.

Reply:
Done. Added pickle-unpickle tests for all combinations as well.